plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -160,49 +170,23 @@ private void registerPluginInstalledEventTask() {
 		}
 	}
 
-	public static SnykExtendedLanguageClient getInstance() {
-		return instance; // we leave instantiation to LSP4e, no lazy construction here
-	}
-
-	public LanguageServer getConnectedLanguageServer() {
-		if (this.ls != null) {
-			return ls;
-		}
-		return super.getLanguageServer();
-	}
-
-	public void triggerScan(IWorkbenchWindow window) {
+	public void triggerScan(String projectPath) {
 		CompletableFuture.runAsync(() -> {
 			if (Preferences.getInstance().getAuthToken().isBlank()) {
 				runSnykWizard();
 			} else {
 				openToolView();
-				try {
-					this.toolView.resetNode(this.toolView.getRoot());
-					if (window == null) {
-						executeCommand(LsCommandID.COMMAND_WORKSPACE_SCAN, new ArrayList<>());
-						return;
-					}
-
-					ISelectionService service = window.getSelectionService();
-					IStructuredSelection structured = (IStructuredSelection) service.getSelection();
-
-					Object firstElement = structured.getFirstElement();
-					IProject project = null;
-					if (firstElement instanceof JavaProject) {
-						project = ((JavaProject) firstElement).getProject();
-					}
-
-					if (firstElement instanceof IProject) {
-						project = (IProject) firstElement;
-					}
 
-					if (project != null) {
-						runForProject(project.getName());
+				if (Preferences.getInstance().getBooleanPref(Preferences.FILTER_DELTA_NEW_ISSUES))
```

Comment:
Food for though: should we add a helper function to Preferences, `isDeltaEnabled()`?